pkg/apiserver/apis/v1alpha3/ipblockusage_types.go

```diff
@@ -0,0 +1,68 @@
+/* Copyright © 2024 Broadcom, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	GroupName    = "nsx.vmware.com"
+	Version      = "v1alpha3"
+	ResourceName = "ipblockusages"
+)
+
+const (
+	External Visibility = "EXTERNAL"
+	Project  Visibility = "PROJECT"
+	Private  Visibility = "PRIVATE"
+)
+
+type Visibility string
+
+type IPAddress struct {
+	Address string `json:"address,omitempty" protobuf:"bytes,1,opt,name=address"`
+	Subnet  string `json:"subnet,omitempty" protobuf:"bytes,2,opt,name=subnet"`
+}
+
+type AllocatedByVPC struct {
+	Count       int         `json:"count,omitempty" protobuf:"bytes,1,opt,name=count"`
+	IPAddresses []IPAddress `json:"ipAddresses,omitempty" protobuf:"bytes,2,opt,name=ipAddresses"`
+	CIDR        string      `json:"cidr,omitempty" protobuf:"bytes,3,opt,name=cidr"`
+	Visibility  Visibility  `json:"visibility,omitempty" protobuf:"bytes,4,opt,name=visibility"`
+}
+
+// IPBlocks contains the used IPAddresses and CIDR.
+type IPBlocks struct {
+	Path      string `json:"path,omitempty" protobuf:"bytes,1,rep,name=path"`
+	Available int    `json:"available,omitempty" protobuf:"bytes,2,rep,name=available"`
+	// AllocatedByVPC contains the cidr and used IP Range.
+	Allocated AllocatedByVPC `json:"allocatedByVpc,omitempty" protobuf:"bytes,3,rep,name=allocatedByVpc"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// IPBlockUsage
+// +k8s:openapi-gen=true
+// +genclient:Namespaced
```

Comment:
If we expose all Tenant scope resources from CCI apiserver, then they should be cluster scoped. Probably let us hold this change until the plan is clear.

Reply:
I would consider adding the API to CCI first, not a Namespace scoped resource in Supervisor.

Reply:
Agreed, if we could expose all IPBlocks(External, Project and Private) info from CCI side, there is no need to do double effort to expose these info in SV any longer.